pkg/systemd/quadlet/quadlet.go

```diff
@@ -1538,3 +1516,37 @@ func handleImageSource(quadletImageName string, serviceUnitFile *parser.UnitFile
 
 	return quadletImageName, nil
 }
+
+func resolveContainerMountParams(containerUnitFile, serviceUnitFile *parser.UnitFile, mount string, names map[string]string) (string, error) {
+	mountType, tokens, err := specgenutilexternal.FindMountType(mount)
+	if err != nil {
+		return "", err
+	}
+
+	// Source resolution is required only for these types of mounts
+	if !(mountType == "volume" || mountType == "bind" || mountType == "glob") {
+		return mount, nil
+	}
+
+	sourceIndex := -1
+	originalSource := ""
+	for i, token := range tokens {
+		kv := strings.SplitN(token, "=", 2)
+		if kv[0] == "source" || kv[0] == "src" {
+			if len(kv) < 2 {
+				return "", fmt.Errorf("source parameter does not include a value")
+			}
+			sourceIndex = i
+			originalSource = kv[1]
+		}
+	}
+
+	resolvedSource, err := handleStorageSource(containerUnitFile, serviceUnitFile, originalSource, names)
+	if err != nil {
+		return "", err
+	}
+	tokens[sourceIndex] = fmt.Sprintf("source=%s", resolvedSource)
+
+	tokens = append([]string{fmt.Sprintf("type=%s", mountType)}, tokens...)
+	return strings.Join(tokens, ","), nil
```

Comment:
This will still fail when the value contains special chars, you must again encode this via the csv lib in order to encode this back into the right format. Consider a a filepath with comma:
`--mount=type=bind,src=/tmp,"dst=/path,1"`

Reply:
Great catch.  Once fixed, can we also add a test case for it?

Reply:
Adding the test case is the first step in writing the fix :)

Reply:
Like Batman, you are not the maintainer we deserve but the one we need :^)

Reply:
Fixed. Wow, making the test work was more complicated than the code fix :)